### PR TITLE
number literals

### PR DESCRIPTION
**Pull request by izulin**
### Context

Adds support to few more number formats.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #579 
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: test/parser/decimal.spec.ts [hunk 1]
```diff
@@ -11,7 +11,11 @@ describe( 'decimal parsing', () => {
       ['+.1', '=+.1'],
       ['+.1', '=+.1+.2'],
       ['=SUM(A1:A4, 0.3, .3)', '=SUM(B1:B4)'],
-      ['.1.4', '=..1']
+      ['.1.4', '=..1'],
+      ['1.', '=1.'],
+      ['1e1', '=1e1'],
+      ['1e+1', '=1e+1'],
+      ['1e-1', '=1e-1'],
```
Comment (wojciechczerniak) [inline test/parser/decimal.spec.ts:18]: Will it work with function as well? And as a criterion?

```suggestion
      ['1e-1', '=1e-1'],
      ['=SUM(A1:A4, 1e-1, .1e+1)', '=IF(A1=1e10, TRUE, FALSE)'],
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added 18 mathematical functions ROMAN, ARABIC, FACT, FACTDOUBLE, COMBIN, COMBINA, GCD, LCM, MROUND, MULTINOMIAL, QUOTIENT, RANDBETWEEN, SERIESSUM, SIGN, SQRTPI, SUMX2MY2, SUMX2PY2, SUMXMY2.
 - Added 63 statistical functions EXPON.DIST, EXPONDIST, FISHER, FISHERINV, GAMMA, GAMMA.DIST, GAMMADIST, GAMMALN, GAMMALN.PRECISE, GAMMA.INV, GAMMAINV, GAUSS, BETA.DIST, BETADIST, BETA.INV, BETAINV, BINOM.DIST, BINOMDIST, BINOM.INV, BESSELI, BESSELJ, BESSELK, BESSELY, CHISQ.DIST, CHISQ.DIST.RT, CHISQ.INV, CHISQ.INV.RT, CHIDIST, CHIINV, F.DIST, F.DIST.RT, F.INV, F.INV.RT, FDIST, FINV, WEIBULL, WEIBULL.DIST, HYPGEOMDIST, HYPGEOM.DIST, T.DIST, T.DIST.2T, T.DIST.RT, T.INV, T.INV.2T, TDIST, TINV, LOGNORM.DIST, LOGNORMDIST, LOGNORM.INV, LOGINV, NORM.DIST, NORMDIST, NORM.S.DIST, NORMSDIST, NORM.INV, NORMINV, NORM.S.INV, NORMSINV, PHI, NEGBINOM.DIST, NEGBINOMDIST, POISSON, POISSON.DIST.
 - Added function aliases mechanism. (#569)
+- Added support for scientific notation. (#579)
```
Comment (wojciechczerniak) [inline CHANGELOG.md:12]: According to https://github.com/handsontable/hyperformula/blob/508e4e4d9961473a35686354f46e3162be2cd53e/docs/guide/types-of-values.md#L10 we already support scientific notation. We're extending the support on formula strings, arguments? Inside criterions `=IF(A1>1e10 ....)`?

Another change is support for `'1.'` and `'=1.'` format.

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (izulin) [reply CHANGELOG.md:12]: 'we already support scientific notation'

that is misleading -- JS is supporting scientific notation, not us

File: test/parser/decimal.spec.ts (hunk 1 above)
Comment (izulin) [reply test/parser/decimal.spec.ts:18]: yes, and those tests do not belong here.

Comment (izulin) [reply test/parser/decimal.spec.ts:18]: I actually don't think this is an useful test. One valid thing to test is SUMIF functionality

Comment (wojciechczerniak) [reply test/parser/decimal.spec.ts:18]: Ok, thanks

**Review by wojciechczerniak — APPROVED** [review]
Awesome, thanks! 🏆

**Pull request merged**